packages/nextjs/components/scaffold-eth/Balance.tsx

```diff
@@ -36,6 +39,10 @@ export const Balance = ({ address, className = "", usdMode }: BalanceProps) => {
     }
   };
 
+  useEffect(() => {
+    queryClient.invalidateQueries({ queryKey });
+  }, [blockNumber, queryClient, queryKey]);
+
```

Comment:
Cool, so this is the `watch` now. 
[![image](https://github.com/scaffold-eth/scaffold-eth-2/assets/2486142/d3290b6f-8090-430e-bb3b-2996e644779f)
](https://wagmi.sh/react/guides/migrate-from-v1-to-v2#removed-watch-property)

(Sorry, just commenting some stuff as I go)